File: 14-longest-common-prefix/longest-common-prefix.swift
import Foundation // Not strictly needed

class Solution {
    func longestCommonPrefix(_ strs: [String]) -> String {
        guard let firstStr = strs.first, !firstStr.isEmpty else {
            return ""
        }
        
        if strs.count == 1 {
            return firstStr
        }

        var prefix = ""
        let firstStrChars = Array(firstStr)

        for i in 0..<firstStrChars.count {
            let charToMatch = firstStrChars[i]
            
            for j in 1..<strs.count {
                let currentStrChars = Array(strs[j])
                if i >= currentStrChars.count || currentStrChars[i] != charToMatch {
                    return prefix
                }
            }
            prefix.append(charToMatch)
        }
        
        return prefix
    }
}

File: 14-longest-common-prefix/longest-common-prefix.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        
        if len(strs) == 1:
            return strs[0]
            
        first_str = strs[0]
        if not first_str:
             return ""

        for i in range(len(first_str)):
            char_to_match = first_str[i]
            for j in range(1, len(strs)):
                current_str = strs[j]
                if i >= len(current_str) or current_str[i] != char_to_match:
                    return first_str[:i]
                    
        return first_str
